pkg/network/dhcp/server/server.go

```diff
@@ -205,6 +205,11 @@ func (h *DHCPHandler) ServeDHCP(p dhcp.Packet, msgType dhcp.MessageType, _ dhcp.
 
 	case dhcp.Request:
 		log.Log.V(4).Info("The request has message type REQUEST")
+		if requestIP, ok := opts[dhcp.OptionRequestedIPAddress]; ok && !net.IP(requestIP).Equal(h.clientIP) {
+			log.Log.Warningf("Client cannot request ip %v, not the assigned ip %v", net.IP(requestIP), h.clientIP)
```

Comment:
I think this warning could be improved, I find it's wording a bit strange.

What do you think about : `Client cannot request ip %v; the IP being advertised by the server is %v` ?

Reply:
Thx for your suggestion, I have pushed a new commit.

Reply:
Let me see what I can do ... 

Care to share your VM spec ? I see a VMI on the attached issue.

Reply:
> Let me see what I can do ...

Yeah, I'm afraid I can't work on a test for this scenario - I don't have windows VMs. 

I would be able to work on top of this for the migration use cases, but that would require more work. 

> 
> Care to share your VM spec ? I see a VMI on the attached issue.